src/renderer/store/matchmaking.store.ts

```diff
@@ -31,88 +48,155 @@ export const matchmakingStore = reactive<{
+async function sendListRequest() {
     matchmakingStore.isLoadingQueues = true;
     matchmakingStore.queueError = undefined;
     try {
         const response = await window.tachyon.request("matchmaking/list");
+        console.log("Tachyon: matchmaking/list:", response.data);
         matchmakingStore.playlists = response.data.playlists;
 
         // Set default selected queue if current selection is not available
         const hasSelectedQueue = matchmakingStore.playlists.some((playlist) => playlist.id === matchmakingStore.selectedQueue);
         if (matchmakingStore.playlists.length > 0 && !hasSelectedQueue) {
             matchmakingStore.selectedQueue = matchmakingStore.playlists[0].id;
         }
+        // Find out of we have the necessary maps for each queue we've been given.
+        matchmakingStore.playlists.forEach((queue) => {
+            if (queue.maps.length > 0) {
+                matchmakingStore.downloadsRequired[queue.id] = { needed: checkIfAnyMapsAreNeeded(queue.maps) };
+            } else {
+                matchmakingStore.downloadsRequired[queue.id] = { needed: true };
+            }
+        });
     } catch (error) {
-        console.error("Failed to fetch available queues", error);
-        matchmakingStore.queueError = "Failed to fetch queues";
+        console.error("Tachyon error: matchmaking/list:", error);
+        matchmakingStore.queueError = "Failed to retrieve available queues";
     } finally {
         matchmakingStore.isLoadingQueues = false;
     }
 }
 
+// This is just to get is a quick idea if any maps will be needed to be downloaded
+function checkIfAnyMapsAreNeeded(maps: { springName: string }[]): boolean {
+    for (const item of maps) {
+        const need = useDexieLiveQueryWithDeps([() => item.springName], () =>
+            db.maps.get(item.springName).then((map) => {
+                if (!map?.isInstalled) {
+                    return true;
+                }
+            })
+        );
+        if (need) return true;
+    }
+    return false;
+}
+
 export function getPlaylistName(id: string): string {
     const playlist = matchmakingStore.playlists.find((playlist) => playlist.id === id);
     return playlist?.name || id;
 }
 
-export const matchmaking = {
-    async startSearch() {
-        try {
-            matchmakingStore.errorMessage = null;
-            matchmakingStore.status = MatchmakingStatus.Searching;
-            await window.tachyon.request("matchmaking/queue", { queues: [matchmakingStore.selectedQueue] });
-        } catch (error) {
-            console.error("Error starting matchmaking:", error);
-            matchmakingStore.errorMessage = "Error: Failed to join the matchmaking queue.";
-            matchmakingStore.status = MatchmakingStatus.Idle;
-        }
-    },
-    stopSearch() {
+async function sendQueueRequest() {
+    try {
+        matchmakingStore.errorMessage = null;
+        matchmakingStore.status = MatchmakingStatus.Searching;
```

Comment:
why set it both here and in `onQueuesJoinedEvent`?

Reply:
Technically, I think my plan when I did that was to have the UI start to treat us as in queue "early" rather than wait for the event, and then if the request failed we would just go back to Idle instead. But considering the possible failure points, I'm going to add a "Requested Join" status that permits us to be between the two.